### PR TITLE
Update caret color of the rich html editor

### PR DESCRIPTION
**Pull request by LunarX**
Make the rich html editor's color follow the theme of the app

Depends on #1958

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt [hunk 1]
```diff
@@ -83,6 +84,7 @@ import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
 import javax.inject.Inject
+import com.google.android.material.R as RMaeterial
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt:87]: Whaet is thaet ?

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt (hunk 1 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageFragment.kt]: You don't like thaet?

**Review by KevinBoulongne — APPROVED** [review]

**Review by tevincent — APPROVED** [review]

File: app/src/main/java/com/infomaniak/mail/utils/Utils.kt [hunk 2]
```diff
@@ -50,6 +50,8 @@ object Utils {
 
     const val SCHEME_SMSTO = "smsto:"
 
+    const val PRIMARY_COLOR_CODE = "--kmail-primary-color"
```
Comment (FabianDevel) [inline app/src/main/java/com/infomaniak/mail/utils/Utils.kt:53]: Wouldn't it make sense to put this in UiUtils instead ?

**Review by FabianDevel — CHANGES_REQUESTED** [review]
The ci crash because of a compilation error on the fDroid flavor

**Comment by LunarX** [comment]
The CI crash is caused by the branch "editor" on which this one relies on. The CI will block this PR from getting merged until it's solved anyway

File: app/src/main/java/com/infomaniak/mail/utils/Utils.kt (hunk 2 above)
Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/utils/Utils.kt:53]: Could very well be, indeed. I moved it there

**Review by FabianDevel — APPROVED** [review]

**Pull request merged**